turbopack/crates/turbopack-ecmascript-runtime/js/src/browser/runtime/base/runtime-base.ts

```diff
@@ -361,7 +361,15 @@ function registerChunk([
   const chunkPath = getPathFromScript(chunkScript)
   for (const [moduleId, moduleFactory] of Object.entries(chunkModules)) {
     if (!moduleFactories[moduleId]) {
-      moduleFactories[moduleId] = moduleFactory
+      if (Array.isArray(moduleFactory)) {
+        let [moduleFactoryFn, otherIds] = moduleFactory
+        moduleFactories[moduleId] = moduleFactoryFn
+        for (const otherModuleId of otherIds) {
+          moduleFactories[otherModuleId] = moduleFactoryFn
+        }
```

Comment:
inlined modules shouldn't have a module factory. Maybe add a throwing module factory instead.

Reply:
this was an optimization we had in the closure compiler.  All imports would get desugared down to either property references on a global object or variable references in the local chunk-group.

This is a global optimization that would be very difficult to make incremental, but it would eliminate a ton of function calls and object allocations from bootstrapping.

Opened https://linear.app/vercel/issue/PACK-4853/globally-flatten-exports for this idea